### ISSUE TITLE
Chinese and Japanese input don't work on iOS (blocked on RN v0.57)

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-mobile#2434]
The normal input methods for users to write in Chinese or in Japanese don't work in our app, on iOS.

You don't need to know any Chinese or Japanese to work on this issue; I've described how to reproduce it below. Ask me here or in chat if something is confusing.

Steps to reproduce:
* In iOS settings, go to "General -> Keyboard -> Keyboards -> Add New Keyboard...". Select "Chinese (Simplified) -> Pinyin -- 10 Key".
* Go to Messages, or some other app. Hit the "globe" key on your keyboard, until you get the keyboard you just added -- it has keys in the middle for "ABC", "DEF", etc., and some keys labeled in Chinese. (You can always hit the "globe" key to get back to your normal keyboard.)
* Hit the keys for M G G A M, in that order. The input box should say "ni hao", which should be selected; and you should have suggestions including "你好".
  * This is "hello" in Chinese. The keyboard has guessed that out of all the letter sequences you could have meant with MNO GHI GHI ABC MNO, the most likely is "ni hao"; or in Chinese characters, "你好".
  * This is how this keyboard is supposed to work.
* Switch to the Zulip app. Narrow to a topic somewhere, and tap the "Reply" field to start composing a message.
* Hit the same five keys. The input box just says "mggam". For me, the suggestions read "is and of said has".
  * It's impossible to type Chinese this way.

The keyboard "Japanese -- Kana" is similarly broken, but the repro is harder to describe starting from no knowledge of Japanese. I'll consider this issue fixed when the Chinese keyboard is working :) -- that'll probably fix the Japanese keyboard too, and I'll file a separate issue if it doesn't.

**Comment by borisyankov** [comment]
This is related:
https://github.com/facebook/react-native/issues/18403

**Comment by hjiang** [comment]
I ran into this problem with the latest iOS app too. This is a video that shows the problem:
http://dropshare.47590191.mtmss.com/1Uh9o3wBA0.mov

All non-latin languages are probably broken.

**Comment by hjiang** [comment]
Seems the upstream issue had been closed. Is there plan to making a new release soon? The current version is unusable for CJK users.

**Comment by borisyankov** [comment]
The fix for the issue is in this commit:
https://github.com/facebook/react-native/commit/892212bad2daadd373f4be241e4cd9889b0a1005

It has not yet landed in a released version of React Native. 
That should be included in 0.57 (but this is a blocker for its release https://github.com/facebook/react-native/issues/20567)


The fix itself is only related to controlled inputs which we are moving away from anyway. But are blocked by another bug (https://github.com/facebook/react-native/issues/18272)

**Comment by gnprice** [comment]
For everyone's enjoyment/education: that upstream commit https://github.com/facebook/react-native/commit/892212bad has a very nice explanation of the diagnosis and the fix.
